google-cloud-firestore/src/main/java/com/google/cloud/firestore/CollectionGroup.java

```diff
@@ -63,6 +65,8 @@
    * parallel. The returned partition cursors are split points that can be used as starting/end
    * points for the query results.
    *
+   * @deprecated This method is deprecated as the SDK has to post-process all results streamed from
```

Comment:
maybe non-ideal, but could we serve the results of `getPartitions(long)` as a stream here, ensuring ordering? It won't actually be streaming of course.

Reply:
We could and I would like more feedback on this decision. I feel it sends the wrong message though to keep the streaming API when it just buffers.

cc @BenWhitehead

Reply:
My concern is that even deprecated, the result is incorrect? I do think streaming sends the wrong message.

Reply:
The result is fetched, then sorted, then released to the user so the behavior should be correct.

Reply:
Discouraging folks I think is good here since the behavioral semantics have changed from what was initially expected. 

I might phrase it differently though
```suggestion
   * @deprecated Please use {@link #getPartitions(long)} instead. All cursors will be loaded before any value will be provided to {@code observer}.
```

We also want to add the `@Deprecated` annotation to the method itself.